### PR TITLE
[ready] auth frontend API

### PR DESCRIPTION
**Pull request by fletcher-greg**
Set up front-end API functions for handling get and post requests

### CODE REVIEW

**Review by SebastianLF — DISMISSED** [review]

File: client/src/services/api.js [hunk 1]
```diff
@@ -1,9 +1,67 @@
...
+
+  async getCachedUser() {
+    try {
+      let { data, status } = await api.get('/auth/getCachedToken', {
+        withCredentials: true
+      })
+
+      // status 204 means there is no cookie, meaning the user is not logged in
+      if (status === 204) {
+        return Promise.resolve({ user: null })
```
Comment (SebastianLF) [inline client/src/services/api.js:52]: Can you precise why status 204 means no cookie?

File: client/src/services/api.js [hunk 2]
```diff
@@ -1,9 +1,67 @@
...
+
+      initApi(data.token)
+      data = (await api.get('/auth/getLoggedUser')).data
+
+      return data
+    } catch (error) {
+      if (error.response.status === 401) {
+        return Promise.resolve({ user: null })
+      }
+    }
```
Comment (SebastianLF) [inline client/src/services/api.js:63]: Why you using try & catch here unlike previous functions?

Comment (lunify) [reply client/src/services/api.js:63]: We are using try catch here following the idea that authorization errors should be the responsibility of the api service. Token validation is one of them for example, as it's not a direct result of user input.
This is a good example of `let` or `const` not being the perfect tool as far as code clarity goes, since it's not clear what the try/catch is trying to do here. I think it would have been a better idea to write something like
```
try {
  var { data, status } = await api.get('/auth/getCachedToken', {
    withCredentials: true
  })

  if (status === 204) // ....
} catch (error) {
  // ....
} 

// extract user authentication and api authorization outside the try catch
initApi(data.token)
data = (await api.get('/auth/getLoggedUser')).data
return data
```
to communicate the intent of specifically catching token validation error

File: client/src/services/api.js (hunk 1 above)
Comment (lunify) [reply client/src/services/api.js:52]: status '204' means the http request's content, syntax and semantics were understood, but "no content" is to be given back (hence the name of the status code). In layman's terms, it's the server answering "Your request was good, but i have nothing to give you back", which is the way we thought was the best answer for this particular situation.
the alternative was a `res.json({ message: 'There is no cached token' })`, but response messages should be displayed to the user, and this one wasn't something we want to show.

Generally, all application layer logic should be answered with 2XX as far as I know, but what I'm sure of is that 4XX errors are reserved for http layer errors as opposed to application layer exceptions.

**Review by SebastianLF — APPROVED** [review]
